### PR TITLE
Implement ghost magnet and speed surge powerup

### PR DESCRIPTION
**Pull request by M1s4kuy4**
Implemented Ghost Magnet and Speed Surge powerups. Upon acquiring the Ghost Magnet, it appears at the pacman`s current location, attracting Ghosts for 5 seconds. The Speed Surge powerup increases speed for 5 seconds when pacman obtained.

### CODE REVIEW

File: Game/GameInfo.pde [hunk 1]
```diff
@@ -74,6 +78,10 @@ public class GameInfo {
   private int playerScore1;
   private int playerScore2;
   private boolean networkSessionClosing;
+  
+  private boolean ghostMagnetActive = false;
+  private float magnetX = 0.0;
+  private float magnetY = 0.0;
```
Comment (NGC0001) [inline Game/GameInfo.pde:86]: Better not to place these three variables inside the `GameInfo` class. The `GameInfo` class should not participate in the game logic.

I would suggest create another SynchronizedItem for the "magnet".
Or, if want a more convenient way, place it inside the `PagePlay` class.

Comment (M1s4kuy4) [reply Game/GameInfo.pde:86]: > I would suggest create another SynchronizedItem for the "magnet".
> Or, if want a more convenient way, place it inside the `PagePlay` class.

Currently I have no idea how Ghosts get the state of Magnet under these two ways.

Comment (NGC0001) [reply Game/GameInfo.pde:86]: Don't worry. You can implement the magnet as such for now. Later I can modify the code.

Comment (M1s4kuy4) [reply Game/GameInfo.pde:86]: Seems like I have successfully modified, I will submit a new version, you can check it again.

**Pull request merged**